extelasticache/node_group_discovery.go

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 Steadybit GmbH
+
+package extelasticache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awshttp "github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/aws/aws-sdk-go-v2/service/elasticache"
+	"github.com/aws/aws-sdk-go-v2/service/elasticache/types"
+	"github.com/rs/zerolog/log"
+	"github.com/steadybit/discovery-kit/go/discovery_kit_api"
+	"github.com/steadybit/discovery-kit/go/discovery_kit_sdk"
+	"github.com/steadybit/extension-aws/config"
+	"github.com/steadybit/extension-aws/utils"
+	"github.com/steadybit/extension-kit/extbuild"
+	"github.com/steadybit/extension-kit/extutil"
+)
+
+type elasticacheReplicationGroupDiscovery struct {
+}
+
+var (
+	_ discovery_kit_sdk.TargetDescriber    = (*elasticacheReplicationGroupDiscovery)(nil)
+	_ discovery_kit_sdk.AttributeDescriber = (*elasticacheReplicationGroupDiscovery)(nil)
+)
+
+func NewElasticacheReplicationGroupDiscovery(ctx context.Context) discovery_kit_sdk.TargetDiscovery {
+	discovery := &elasticacheReplicationGroupDiscovery{}
+	return discovery_kit_sdk.NewCachedTargetDiscovery(discovery,
+		discovery_kit_sdk.WithRefreshTargetsNow(),
+		discovery_kit_sdk.WithRefreshTargetsInterval(ctx, time.Duration(config.Config.DiscoveryIntervalElasticacheReplicationGroup)*time.Second),
+	)
+}
+
+func (r *elasticacheReplicationGroupDiscovery) Describe() discovery_kit_api.DiscoveryDescription {
+	return discovery_kit_api.DiscoveryDescription{
+		Id: elasticacheNodeGroupTargetId,
+		Discover: discovery_kit_api.DescribingEndpointReferenceWithCallInterval{
+			CallInterval: extutil.Ptr(fmt.Sprintf("%ds", config.Config.DiscoveryIntervalElasticacheReplicationGroup)),
+		},
+	}
+}
+
+func (r *elasticacheReplicationGroupDiscovery) DescribeTarget() discovery_kit_api.TargetDescription {
+	return discovery_kit_api.TargetDescription{
+		Id:       elasticacheNodeGroupTargetId,
+		Label:    discovery_kit_api.PluralLabel{One: "Elasticache replication group", Other: "Elasticache replication groups"},
+		Category: extutil.Ptr("cloud"),
+		Version:  extbuild.GetSemverVersionStringOrUnknown(),
+		Icon:     extutil.Ptr(elasticacheIcon),
+		Table: discovery_kit_api.Table{
+			Columns: []discovery_kit_api.Column{
+				{Attribute: "steadybit.label"},
```

Comment:
at the end when we return the target, the label is the concatenation of the replication group ID + nodegroup ID.

Since the attack is occuring on nodegroup but the replication group id is still needed, I prefer to put that in the steadybit.label, because the node group id is not really unique (eg: "0001" etc).